rsocket-ipc-graphql/src/main/java/io/rsocket/graphql/GraphQLServerRequestStream.java

```diff
@@ -0,0 +1,85 @@
+package io.rsocket.graphql;
+
+import graphql.ExecutionInput;
+import graphql.ExecutionResult;
+import graphql.GraphQL;
+import graphql.GraphQLError;
+import graphql.execution.ExecutionId;
+import graphql.execution.instrumentation.Instrumentation;
+import graphql.schema.GraphQLSchema;
+import io.netty.buffer.ByteBuf;
+import io.rsocket.ipc.Functions;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import org.dataloader.DataLoaderRegistry;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+@SuppressWarnings("unchecked")
+class GraphQLServerRequestStream implements Functions.RequestStream<GraphQLRequest, Object> {
+  private final DataLoaderRegistry registry;
+  private final Instrumentation instrumentation;
+  private final GraphQLSchema graphQLSchema;
+
+  GraphQLServerRequestStream(
+      DataLoaderRegistry registry, Instrumentation instrumentation, GraphQLSchema graphQLSchema) {
+    this.registry = registry;
+    this.instrumentation = instrumentation;
+    this.graphQLSchema = graphQLSchema;
+  }
+
+  @Override
+  public Flux<Object> apply(GraphQLRequest request, ByteBuf byteBuf) {
+    try {
+      ExecutionInput.Builder builder =
+          ExecutionInput.newExecutionInput()
+              .query(request.getQuery())
+              .operationName(request.getOperationName())
+              .variables(request.getVariables())
+              .context(byteBuf)
+              .executionId(ExecutionId.generate());
+
+      if (registry != null) {
+        builder.dataLoaderRegistry(registry);
+      }
+
+      ExecutionInput executionInput = builder.build();
+
+      CompletableFuture<ExecutionResult> result =
```

Comment:
Should this work be deferred? In case of error etc, it seems wasteful to start this outside of an established subscription.

Reply:
Or is that executeAsync?

Reply:
its asynchronous